app/models/post.rb

```diff
@@ -546,12 +546,17 @@ def hide!(post_action_type_id, reason = nil, custom_message: nil)
     self.hidden_at = Time.zone.now
     self.hidden_reason_id = reason
     self.skip_unique_check = true
-    save!
 
-    Topic.where(
-      "id = :topic_id AND NOT EXISTS(SELECT 1 FROM POSTS WHERE topic_id = :topic_id AND NOT hidden)",
-      topic_id: topic_id
-    ).update_all(visible: false)
+    Post.transaction do
+      save!
+
+      Topic.where(
+        "id = :topic_id AND NOT EXISTS(SELECT 1 FROM POSTS WHERE topic_id = :topic_id AND NOT hidden)",
+        topic_id: topic_id
+      ).update_all(visible: false)
```

Comment:
Don't have to make any changes, just a comment:

I understand this chunk is just reformatted but my brain had to bend a bit to parse 

```
Topic.where(
        "id = :topic_id AND NOT EXISTS(SELECT 1 FROM POSTS WHERE topic_id = :topic_id AND NOT hidden)",
        topic_id: topic_id
      )
```

Would be nice if we can describe what it does somehow in english like "topic with hidden posts" or sth. (or having one of my fav scopes :D)